src/Microsoft.Management.Deployment/Helpers.cpp

```diff
@@ -0,0 +1,97 @@
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include <wil/resource.h>
+#include <wil/win32_helpers.h>
+#include <winrt/Windows.Security.Authorization.AppCapabilityAccess.h>
+#include <appmodel.h>
+#include <Helpers.h>
+
+namespace winrt::Microsoft::Management::Deployment::implementation
+{
+    std::optional<DWORD> GetCallerProcessId()
+    {
+        RPC_STATUS rpcStatus = RPC_S_OK;
+        RPC_CALL_ATTRIBUTES callAttributes = {};
+        callAttributes.Version = RPC_CALL_ATTRIBUTES_VERSION;
+        callAttributes.Flags = RPC_QUERY_CLIENT_PID;
+        rpcStatus = RpcServerInqCallAttributes(nullptr, &callAttributes);
+
+        if ((rpcStatus != RPC_S_NO_CALL_ACTIVE) &&
+            !((rpcStatus == RPC_S_OK) && HandleToULong(callAttributes.ClientPID) == GetCurrentProcessId()))
+        {
+            DWORD callerProcessId = HandleToULong(callAttributes.ClientPID);
+            return callerProcessId;
+        }
+        return {};
+    }
+
+    std::wstring GetStringForCapability(Capability capability)

```

Comment:
Better to return a `std::wstring_view` in these cases to prevent an unnecessary allocation.